### PR TITLE
Feature: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Akorede3133**
## In this PR, the following were affected;

* Created class `Person` with the following:
    - [ ] Instance vars: @id, @name, and @age.
    - [ ] Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
    - [ ] Getters for @id, @name, and @age.
    - [ ] Setters for @name and @age.
    - [ ] Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
    - [ ] Public method can_use_services? that returns true if person is of age or if they have permission from parents.

* Created class `Student` with the following:
    - [ ]   Inherits from Person.
    - [ ]     Constructor extends parent's constructor by adding @classroom and a parameter for it.
    - [ ]     Method play_hooky that returns "¯\(ツ)/¯".

* Created class `Teacher` with the following:
- [ ]  Inherits from Person.
- [ ]     Constructor extends parent's constructor by adding @specialization and a parameter for it.
- [ ]     Override can_use_services? so it always returns true.

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Akorede3133,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that does not have a descriptive README.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(id, age, name = 'Unknown', parent_permission: true)
+    @id = id
```
Comment (DammyShittu) [inline person.rb]: - [ ] We can auto-generate the `id` of a person so we don't have to pass it in everytime we create a person. Then you can remove the `id` from the list of parameters.

  ```
  def initialize(age, name = 'Unknown', parent_permission: true)
    @id = rand(1...1000)
    ...
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(id, age, classroom, name = 'Uknown', parent_permission: true)
+    super(id, age, name, parent_permission)
```
Comment (DammyShittu) [inline student.rb]: - [ ] Kindly remove the `id` from here.

- [ ] The `parent_permission` argument should be assigned the correct way in the `super` method argument

  ```
  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    ...
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(id, age, specialization, name = 'Uknown', parent_permission: true)
+    super(id, age, name, parent_permission)
```
Comment (DammyShittu) [inline teacher.rb]: - [ ] Kindly remove the `id` from here. It is now auto-generated by the app.

- [ ] The `parent_permission` argument should be assigned the correct way in the `super` method argument

  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    ...
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Akorede3133,

## Highlights
✔️ Private method feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Akorede3133,

✔️ Classes are created in different files based on convention

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**